### PR TITLE
Fixes small bug in dompurify types

### PR DESCRIPTION
**Pull request by fstanis**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

File: types/dompurify/index.d.ts [hunk 1]
```diff
@@ -83,13 +83,13 @@ declare namespace DOMPurify {
 
     interface SanitizeElementHookEvent {
         tagName: string;
-        allowedTags: string[];
+        allowedTags: { [key: string]: boolean };
```
Comment (peterblazejewicz) [inline types/dompurify/index.d.ts]: as you checked tests option, can we amend the test file to cover that change? Thanks!
```ts
//test the 'SanitizeElementHookEvent' type is publicly accessible.
dompurify.addHook('uponSanitizeElement', (currentNode: Element, event: DOMPurify.SanitizeElementHookEvent) => {
    if (currentNode.nodeName && currentNode.nodeName.match(/^\w+-\w+$/) && !event.allowedTags[event.tagName]) {
        event.allowedTags[event.tagName] = true;
    }
});
```

File: types/dompurify/index.d.ts [hunk 2]
```diff
@@ -83,13 +83,13 @@ declare namespace DOMPurify {
...
     }
 
     interface SanitizeAttributeHookEvent {
         attrName: string;
         attrValue: string;
         keepAttr: boolean;
-        allowedAttributes: string[];
+        allowedAttributes: { [key: string]: boolean };
```
Comment (peterblazejewicz) [inline types/dompurify/index.d.ts]: same, as you checked test option, can we amend the test that covers that hook?

```ts
//test the 'SanitizeAttributeHookEvent' type is publicly accessible.
dompurify.addHook('uponSanitizeAttribute', (currentNode: Element, event: DOMPurify.SanitizeAttributeHookEvent) => {
    if (event.attrName && event.attrName.match(/^\w+-\w+$/) && !event.allowedAttributes[event.attrName]) {
        event.allowedAttributes[event.attrName] = true;
    }
});
```
ref:
https://github.com/cure53/DOMPurify/blob/da4663394f59cfbd313996b858cf499689be76c3/test/test-suite.js#L289-L300
Thanks!

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]

File: types/dompurify/index.d.ts (hunk 1 above)
Comment (fstanis) [reply types/dompurify/index.d.ts]: Done.

File: types/dompurify/index.d.ts (hunk 2 above)
Comment (fstanis) [reply types/dompurify/index.d.ts]: Done.

**Review by peterblazejewicz — APPROVED** [review]
LGTM
Up to maintainers!
Thanks for contribution!

**Review by peterblazejewicz — APPROVED** [review]

**Review by Exigerr — APPROVED** [review]

**Pull request merged**